ecommerce/courses/utils.py

```
@@ -33,6 +38,71 @@ def get_course_info_from_catalog(site, course_key):
     return course_run
 
 
+def get_course_catalogs(site, resource_id=None):
+    """
+    Get details related to course catalogs from Catalog Service.
+
+    Arguments:
+        site (Site): Site object containing Site Configuration data
+        resource_id (int or str): Identifies a specific resource to be retrieved
+
+    Returns:
+        dict: Course catalogs received from Course Catalog API
+
+    """
+    resource = 'catalogs'
+    base_cache_key = 'catalog.api.data'
+
+    cache_key = '{}.{}'.format(base_cache_key, resource_id) if resource_id else base_cache_key
+    cache_key = hashlib.md5(cache_key).hexdigest()
+    cached = cache.get(cache_key)
+    if cached:
+        return cached
+
+    api = site.siteconfiguration.course_catalog_api_client
+    try:
+        endpoint = getattr(api, resource)
+        response = endpoint(resource_id).get()
+
+        if resource_id:
+            results = response
+        else:
+            results = traverse_pagination(response, endpoint)
+    except:  # pylint: disable=bare-except
+        logger.exception('Failed to retrieve data from the Course Discovery API.')
```

This method should throw an exception not log an error.  The caller should try / catch this call and do what is appropriate.  See get_course_info_from_catalog.